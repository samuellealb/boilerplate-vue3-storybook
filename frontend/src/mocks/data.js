export const jobs = {
  data: [
    {
      "id": 1,
      "attributes": {
        "Title": "Mocked Job",
        "Description": "Lorem ipsum dolor sit amet",
        "categories": {
          "data": [
            {
              "id": 2,
              "attributes": {
                  "Title": "Consectetur",
              }
            },
            {
              "id": 3,
              "attributes": {
                "Title": "Dolorem",
              }
            }
          ]
        },
        "cover": {
          "data": {
            "id": 1,
            "attributes": {
              "alternativeText": "alternative text",
              "caption": "ima-white-logo.png",
              "url": "https://picsum.photos/400/300",
            }
          }
        }
      }
    }
  ] 
}