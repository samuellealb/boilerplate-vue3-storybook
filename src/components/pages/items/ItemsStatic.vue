<template>
  <default-layout>
    <div class="items">
      <h1>Items</h1>
      <item-list :items="items" />
    </div>
  </default-layout>
</template>

<script>
import './items.css'
import ItemList from '@/components/organisms/item-list/ItemList.vue';
import DefaultLayout from "@/layouts/DefaultLayout.js"

export default {
  name: 'ItemsStatic',
  props: {
    items: {
      type: Array,
      default: () => [],
      required: true
    }
  },
  components: {
    ItemList,
    DefaultLayout
  },
}
</script>