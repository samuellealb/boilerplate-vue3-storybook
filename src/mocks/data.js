export const items = [
  {
    id: 1,
    attributes: {
      Name: 'Itamar Assumpção'
    }
  },
  {
    id: 2,
    attributes: {
      Name: 'Clara Nunes' 
    }
  },
  {
    id: 3,
    attributes: {
      Name: 'Tim Maia'
    }
  }
]