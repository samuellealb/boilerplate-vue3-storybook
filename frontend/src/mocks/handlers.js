import { rest } from 'msw'
import { jobs } from './data'

const apiURL = import.meta.env.VITE_API_URL

export const handlers = [
  rest.get(`/jobs`, (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json(jobs)
    )
  })
] 