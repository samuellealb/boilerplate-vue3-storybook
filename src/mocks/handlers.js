import { rest } from 'msw'
import { items } from './data'

const loadedItems = {data: [...items]} 

export const handlers = [
  rest.get(`/api/items`, (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json(loadedItems)
    )
  }),

  rest.delete(`/api/items/:itemId`, (req, res, ctx) => {
    const { itemId } = req.params
    const item = loadedItems.data.find(item => item.id === +itemId);
    const index = loadedItems.data.findIndex(item => item.id === +itemId);
    loadedItems.data.splice(index, 1);
    return res(
      ctx.status(200),
      ctx.json(item)
    )
  })
] 