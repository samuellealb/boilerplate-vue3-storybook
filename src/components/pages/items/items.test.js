import { mount } from '@vue/test-utils'
import { getAxios } from '@/utils/get-axios.js'
import { vi, test, expect, describe } from 'vitest'
import ItemsStatic from './ItemsStatic.vue'
import ItemsPage from './Items.vue'
import { items } from '@/mocks/data'
const axios = getAxios()

describe('mount component', () => {
    
  test('renders empty list', () => {
    expect(ItemsStatic).toBeTruthy()
    const noItem = mount(ItemsStatic, { props: { items: [] } })
    expect(noItem.find('.item-list').findAll('h2').length).toEqual(1)
    expect(noItem.html()).toMatchSnapshot()
  })
  
  test('renders list with three items', () => {
    expect(ItemsStatic).toBeTruthy()
    const threeItems = mount(ItemsStatic, { props: { items } })
    expect(threeItems.find('.item-list').findAll('.item-card').length).toEqual(3)
    expect(threeItems.html()).toMatchSnapshot()
  })
  
})

describe("API Calls", () => {

  describe("fetchItems being called successufuly", () => {
    test("should return items list", async () => {    
      vi.spyOn(axios, 'get').mockResolvedValue(items)
      mount(ItemsPage, { props: { items: [] } })
      expect(axios.get).toHaveBeenCalledTimes(1)
      expect(axios.get).toHaveBeenCalledWith('/api/items')
    });
  });

  describe("deleteItem being called successufuly", () => {
    test("should return deleted item", async () => {    
      vi.spyOn(axios, 'delete')
      const wrapper = mount(ItemsPage, { props: { items } })
      // wrapper.find('.item-card:nth-of-type(2) button').trigger('click')
      // expect(axios.delete).toHaveBeenCalledTimes(1)
      // expect(axios.delete).toHaveBeenCalledWith('/api/items/2')
    });
  });

});
