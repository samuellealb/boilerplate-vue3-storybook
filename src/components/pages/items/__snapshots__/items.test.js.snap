// Vitest Snapshot v1

exports[`mount component > renders empty list 1`] = `
"<div class=\\"items\\">
  <h1>Items</h1>
  <div class=\\"item-list\\">
    <h2>No items loaded =/</h2>
  </div>
</div>"
`;

exports[`mount component > renders list with three items 1`] = `
"<div class=\\"items\\">
  <h1>Items</h1>
  <div class=\\"item-list\\">
    <div class=\\"item-card\\">
      <div class=\\"item-card__title\\">Itamar Assumpção</div>
      <div class=\\"item-card__button\\"><button class=\\"simple-button simple-button--default simple-button--small\\">Delete</button></div>
    </div>
    <div class=\\"item-card\\">
      <div class=\\"item-card__title\\">Clara Nunes</div>
      <div class=\\"item-card__button\\"><button class=\\"simple-button simple-button--default simple-button--small\\">Delete</button></div>
    </div>
    <div class=\\"item-card\\">
      <div class=\\"item-card__title\\">Tim Maia</div>
      <div class=\\"item-card__button\\"><button class=\\"simple-button simple-button--default simple-button--small\\">Delete</button></div>
    </div>
  </div>
</div>"
`;
